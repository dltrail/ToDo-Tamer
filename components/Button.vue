<script setup lang="ts">
  interface Props {
    iconName?: string;
    isDisabled?: boolean;
  }
  const props = defineProps<Props>()
</script>

<template>
  <button data-testid="button" class="button" :disabled="isDisabled" role="button">
    <UIcon data-testid="button-icon" v-if="iconName" :name="iconName" class="button-icon" />
    <span><slot /></span>
  </button>
</template>

<style scoped>
.button {
  @apply text-white px-4 text-xs border p-2 border-[var(--primary-button-color)] bg-white rounded-md text-[var(--primary-button-color)] hover:opacity-90 align-middle
 py-2 flex gap-2 justify-center items-center;
}

.button-icon {
  @apply w-4 h-4 text-[var(--primary-button-color)] align-middle;
}

:disabled {
  @apply pointer-events-none opacity-50;
}
</style>
