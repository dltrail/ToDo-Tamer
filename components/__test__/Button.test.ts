import { describe, test, expect } from "vitest";
import { mount } from "@vue/test-utils";
import "@testing-library/jest-dom"; 
import Button from "../Button.vue"; 

describe("Button", () => {
  test("renders the button with correct button text", () => {
    const wrapper = mount(Button, {
      slots: {
        default: "Button text",
      },
    });
    expect(wrapper.find('span').text()).toBe('Button text');
  });

  test("button is rendered", () => {
    const wrapper = mount(Button, {
      slots: {
        default: "Button text",
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("button icon is rendered correctly", () => {
    const wrapper = mount(Button, {
      props: {
        iconName: "i-heroicons-trash",
      },
      
    });
    const icon = wrapper.find('[data-testid="button-icon"]')
    expect(icon.classes()).toEqual(expect.arrayContaining(['button-icon']))
  });

  test("button icon is not rendered if there is no icon name", () => {
    const wrapper = mount(Button, {
      props: {
        iconName: "",
      },
    });
    const icon = wrapper.find('[data-testid="button-icon"]')
  expect(icon.exists()).toBe(false)
  });
});
