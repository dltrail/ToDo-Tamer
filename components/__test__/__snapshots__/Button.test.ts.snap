// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > button is rendered 1`] = `
"<button data-v-b534b83e="" data-testid="button" class="button" role="button">
  <!--v-if--><span data-v-b534b83e="">Button text</span>
</button>"
`;
